### DISCUSSION TITLE
minio 20221024183507.0.0-1 not supporting boto3 client with large files like 100MB

### DISCUSSION THREAD

**Discussion by suhailsid** [minio/minio#16588] [Q&A]
hello All,

I was using minio verion - minio-20210527220631.0.0-1.x86_64 earlier to access on https instread port 9000 we have configured reverse nginx proxy , we are using python application which use boto3 (v1.17.107 with botocore v1.20.107)client to access minio buckets and which was working with this version perfectly.
the issue starting after we upgrade minio server fromminio-20210527220631.0.0-1.x86_64 to version minio-20221024183507.0.0-1.x86_64  now we have console also enable to run with ssl on 443 with some changes in revers proxy now the issue is after upgrade boto3 client not able to get and copy the files in same bucket in different location , they can do this with small size file but when they use files size more than 100 MB operation getting failed and boto3 operation get stuck in loop.

this is my /etc/default/minio config file 

MINIO_VOLUMES="/opt/SP/minio"
MINIO_ROOT_USER=minio
MINIO_ROOT_PASSWORD=minio
MINIO_OPTS=" --certs-dir /home/minio/.minio/certs --address ':9000' --console-address :8443"
MINIO_BROWSER_REDIRECT_URL=https://minio-console.example.com
MINIO_SERVER_URL=https://minio.example.com

**Comment by klauspost** [comment]
We need some more info, like the error boto3 is experiencing. Or the very least `mc admin trace ALIAS` output of the queries.

Secondly this is an outdated version. We only support the latest.

**Comment by suhailsid** [comment]
I tried this with latest minio version and result are same I move object from one folder to another folder in same bucket getting stuck using boto3 script.

one more issue I am observing in new minio version is when i open browser bucket its show always loading nothing inside bucket I can see via console browser. 

![image](https://user-images.githubusercontent.com/35805910/218684485-8b704980-a491-4f30-8a4c-bd73160097e0.png)

**Reply by klauspost** [reply]
You are probably using a load balancer/reverse proxy that isn't configured for web sockets. This causes buckets to not load.

**Comment by suhailsid** [comment]
this is really strange issue, Minio server hosted on RHEL7 server and after minio upgrade i did the "mcli admin config reset myminio/ api" and restart server network services using "service network restart" and it worked, but after some month the same issue occurred again, and when I am doing same steps, it's not working anymore.  
below code we are using to do the action, mcli debug and all trace also not shwoing anything on this issue 
``
import logging
import sys
import boto3
import os
env = sys.argv[1]
print('enviroment set to...', env)

from vodaflowutils import minio_client
def get_minio_client(
        p_minio_conf: dict,
        p_logger: logging.Logger = logging.getLogger(),
        p_log_prefix: str = "===> [get_minio_client]: "
        ):
    """ Returns a MinIO client object. """

    err_msg = ""
    minio_pw = ""
    p_target_folder_path = os.getcwd()
    p_logger.info(p_log_prefix + "Creating MinIO client...")

    minio_pw = p_minio_conf["aws_secret_access_token_id"]

    try:
        mc = minio_client.MinioClient(
            p_endpoint_url=p_minio_conf["endpoint_url"],
            p_signature_version=p_minio_conf["signature_version"],
            p_aws_access_key_id=p_minio_conf["aws_access_key_id"],
            p_minio_bucket_name=p_minio_conf["bucket_name"],
            p_aws_secret_access_key=minio_pw,
            p_logger=p_logger,
            p_mozart_in_folder_path=p_minio_conf[
                    "input_bucket_base_path_name"],
            p_mozart_out_folder_path=p_minio_conf[
                    "output_bucket_base_path_name"],
            p_mozart_temp_folder_path=p_minio_conf[
                    "temp_bucket_base_path_name"]
            )
        p_logger.info(p_log_prefix + "Successfully created MinIO client.")

        derived_extns = ['a9..', 'a0..', 'gz', 'csv', 'avro']
        derived_stem = '*'
        object_list, derived_stem, derived_extns \
            = mc. \
            get_object_list_with_stem(p_file_stem=derived_stem,
                                      p_lower_valid_extensions=
                                      derived_extns)

        # endpoint_url = p_minio_conf["endpoint_url"]
        # aws_access_key_id = p_minio_conf["aws_access_key_id"]
        # aws_secret_access_key = p_minio_conf["aws_secret_access_token_id"]
        # p_logger.info(f"Trying to connect to Minio - {endpoint_url}")
        #
        # s3_storage_service = boto3.resource(
        #     's3',
        #     endpoint_url=endpoint_url,
        #     aws_access_key_id= aws_access_key_id,
        #     aws_secret_access_key= aws_secret_access_key,
        #     config= p_minio_conf["signature_version"],
        #     verify=False)
        # p_logger.info(f"Connected to Minio - {endpoint_url}!")
        # object_list = s3_storage_service.list_objects(p_minio_conf["bucket_name"])
        for object in object_list:
            print('printing object from bucket....')
            print(object)
        mc.download_files(p_minio_object_names=object_list,
                                            p_landing_zone_path=str(
                                                p_target_folder_path) + "/")
        print("Downloaded Files Successfully!")
        p_logger.info(p_log_prefix + "Downloaded Files Successfully!")
        print("Moving Files to Temp...")

        mc.move_files_to_temp(p_minio_object_names=object_list)
        print("Temp Files moved Successfully!")
        p_logger.info(p_log_prefix + "done")
    except Exception as e:
        err_msg = "Error creating MinIO client."
        p_logger.critical(p_log_prefix + err_msg)
        raise e

    return

if env.lower() == 'dev':
    p_minio_conf = {
 "endpoint_url": "https://minio.server.net/",
 "signature_version": "s3v4",
 "aws_access_key_id": "access-key",
"aws_secret_access_token_id": "secret-key",
 "bucket_name": "data-lake-dev",
 "input_bucket_base_path_name": "dlake/In/",
 "output_bucket_base_path_name": "dlake/Out/",
 "temp_bucket_base_path_name": "dlake/Temp/",
 "integration_test_folder_path": "dlake/integration_tests/"
}

``

**Reply by suhailsid** [reply]
@klauspost  & @harshavardhana pls help this is really strange issue

**Reply by klauspost** [reply]
I don't see any usable information above and we do not perform code/setup reviews as a community service. See our [subscriptions](https://min.io/pricing) for more detailed support.
